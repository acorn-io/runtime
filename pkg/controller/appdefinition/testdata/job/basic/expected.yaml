kind: Namespace
apiVersion: v1
metadata:
  name: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    pod-security.kubernetes.io/enforce: baseline

---
kind: Job
apiVersion: batch/v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/job-name": "oneimage"
    "acorn.io/managed": "true"
    "acorn.io/root-prefix": "app-name"
spec:
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/root-prefix": "app-name"
        "acorn.io/job-name": "oneimage"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"build":{"context":".","dockerfile":"Dockerfile"},"image":"image-name","ports":[{"port":80,"protocol":"http","targetPort":81}],"sidecars":{"left":{"image":"foo","ports":[{"port":90,"protocol":"tcp","targetPort":91}]}}}'
    spec:
      imagePullSecrets:
        - name: oneimage-pull-1234567890ab
      restartPolicy: Never
      automountServiceAccountToken: false
      enableServiceLinks: false
      terminationGracePeriodSeconds: 5
      containers:
        - name: oneimage
          image: "image-name"
          readinessProbe:
            tcpSocket:
              port: 81
          ports:
            - containerPort: 81
              protocol: "TCP"
          terminationMessagePath: "/run/secrets/output"
        - name: left
          image: "foo"
          readinessProbe:
            tcpSocket:
              port: 91
          ports:
            - containerPort: 91
              protocol: "TCP"
          terminationMessagePath: "/run/secrets/output"
---
kind: Secret
apiVersion: v1
metadata:
  name: oneimage-pull-1234567890ab
  namespace: app-created-namespace
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
type: "kubernetes.io/dockerconfigjson"
data:
  ".dockerconfigjson": eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
---
kind: AppInstance
apiVersion: internal.acorn.io/v1
metadata:
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
status:
  namespace: app-created-namespace
  appSpec:
    jobs:
      oneimage:
        sidecars:
          left:
            image: "foo"
            ports:
              - port: 90
                targetPort: 91
                protocol: tcp
        ports:
          - port: 80
            targetPort: 81
            protocol: http
        image: "image-name"
        build:
          dockerfile: "Dockerfile"
          context: "."
  conditions:
    - type: defined
      reason: Success
      status: "True"
      success: true
