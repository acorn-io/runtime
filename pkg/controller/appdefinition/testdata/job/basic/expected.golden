`apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJnaGNyLmlvIjp7ImF1dGgiOiJPZz09In0sImluZGV4LmRvY2tlci5pbyI6eyJhdXRoIjoiT2c9PSJ9fX0=
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: oneimage-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    acorn.io/app-generation: "0"
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/job-name: oneimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: oneimage
  namespace: app-created-namespace

---
apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    acorn.io/app-generation: "0"
    acorn.io/config-hash: ""
    apply.acorn.io/prune: "false"
    apply.acorn.io/update: "true"
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/job-name: oneimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: oneimage
  namespace: app-created-namespace
spec:
  backoffLimit: 1000
  template:
    metadata:
      annotations:
        acorn.io/config-hash: ""
        acorn.io/container-spec: '{"build":{"context":".","dockerfile":"Dockerfile"},"image":"image-name","metrics":{},"ports":[{"port":80,"protocol":"http","targetPort":81}],"probes":null,"sidecars":{"left":{"image":"foo","metrics":{},"ports":[{"port":90,"protocol":"tcp","targetPort":91}],"probes":null}}}'
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/job-name: oneimage
        acorn.io/managed: "true"
        acorn.io/project-name: app-namespace
    spec:
      containers:
      - env:
        - name: ACORN_EVENT
          value: create
        image: image-name
        name: oneimage
        ports:
        - containerPort: 81
          protocol: TCP
        readinessProbe:
          tcpSocket:
            port: 81
        resources: {}
        volumeMounts:
        - mountPath: /run/secrets
          name: acorn-job-output-helper
      - env:
        - name: ACORN_EVENT
          value: create
        image: foo
        name: left
        ports:
        - containerPort: 91
          protocol: TCP
        readinessProbe:
          tcpSocket:
            port: 91
        resources: {}
        volumeMounts:
        - mountPath: /run/secrets
          name: acorn-job-output-helper
      - command:
        - /usr/local/bin/acorn-job-helper-init
        env:
        - name: ACORN_EVENT
          value: create
        image: ghcr.io/acorn-io/runtime:main
        imagePullPolicy: IfNotPresent
        name: acorn-job-output-helper
        resources: {}
        volumeMounts:
        - mountPath: /run/secrets
          name: acorn-job-output-helper
      enableServiceLinks: false
      imagePullSecrets:
      - name: oneimage-pull-1234567890ab
      restartPolicy: Never
      serviceAccountName: oneimage
      terminationGracePeriodSeconds: 5
      volumes:
      - emptyDir:
          medium: Memory
          sizeLimit: 1M
        name: acorn-job-output-helper
status: {}

---
apiVersion: internal.acorn.io/v1
kind: ServiceInstance
metadata:
  annotations:
    acorn.io/app-generation: "0"
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/job-name: oneimage
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.oneimage
  name: oneimage
  namespace: app-created-namespace
spec:
  appName: app-name
  appNamespace: app-namespace
  container: oneimage
  default: false
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/job-name: oneimage
    acorn.io/managed: "true"
  ports:
  - port: 80
    protocol: http
    targetPort: 81
  - port: 90
    protocol: tcp
    targetPort: 91
status: {}

---
apiVersion: internal.acorn.io/v1
kind: AppInstance
metadata:
  creationTimestamp: null
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
status:
  appImage:
    buildContext: {}
    id: test
    imageData: {}
    vcs: {}
  appSpec:
    jobs:
      oneimage:
        build:
          context: .
          dockerfile: Dockerfile
        image: image-name
        metrics: {}
        ports:
        - port: 80
          protocol: http
          targetPort: 81
        probes: null
        sidecars:
          left:
            image: foo
            metrics: {}
            ports:
            - port: 90
              protocol: tcp
              targetPort: 91
            probes: null
  appStatus:
    jobs:
      oneimage: {}
  columns: {}
  conditions:
    reason: Success
    status: "True"
    success: true
    type: defined
  defaults: {}
  namespace: app-created-namespace
  staged:
    appImage:
      buildContext: {}
      imageData: {}
      vcs: {}
  summary: {}
`
